drape/gl_functions.cpp

```diff
@@ -1050,6 +1063,15 @@ void GLFunctions::glTexSubImage2D(int x, int y, int width, int height, glConst l
                                   glConst pixelType, void const * data)
 {
   ASSERT_NOT_EQUAL(CurrentApiVersion, dp::ApiVersion::Invalid, ());
+  /*
+  if (CurrentApiVersion == dp::ApiVersion::OpenGLES2 &&
+      layout == GL_ALPHA &&
+      pixelType == GL_UNSIGNED_BYTE)
+  {
+  	// On Linux Wayland GL_INVALID_OPERATION happens here in the
```

Comment:
Does somebody have an idea why this is?

Reply:
No, but looks like you should check dp::UnpackFormat function, where layout and pixelType vars are initialized. There are _magic_ comments like:
```
// On OpenGL ES3 GLAlpha is not supported, we use GLRed instead.
```

So you can experiment with some other constants combination, like GLRed (GL_RED).

Reply:
Thanks for the tip @vng.
At first glance, using `GLRed` does seem to help the `GL_INVALID_OPERATION` here.
And it turns out that the blurryness is possibly independent from that `GL_INVALID_OPERATION` here,
it is caused by something else, somewhere else.